tsl/src/gapfill/exec.c

```diff
@@ -178,6 +189,257 @@ gapfill_state_create(CustomScan *cscan)
 	return (Node *) state;
 }
 
+/*
+ * lookup operator name in pg_operator
+ *
+ * returns NULL for unknown operator
+ */
+static char *
+get_oprname(Oid oprid)
+{
+	HeapTuple	opertup;
+	char	   *result = NULL;
+
+	if (!OidIsValid(oprid))
+		return result;
+
+	opertup = SearchSysCache1(OPEROID, ObjectIdGetDatum(oprid));
+	if (HeapTupleIsValid(opertup))
+	{
+		Form_pg_operator operform = (Form_pg_operator) GETSTRUCT(opertup);
+
+		result = pstrdup(NameStr(operform->oprname));
+
+		ReleaseSysCache(opertup);
+	}
+
+	return result;
+}
+
+/*
+ * lookup cast func oid in pg_cast
+ *
+ * throws an error if no cast can be found
+ */
+static Oid
+get_castfunc(Oid source, Oid target)
+{
+	Oid			result = InvalidOid;
+	HeapTuple	casttup;
+
+	casttup = SearchSysCache2(CASTSOURCETARGET, ObjectIdGetDatum(source), ObjectIdGetDatum(target));
+	if (HeapTupleIsValid(casttup))
+	{
+		Form_pg_cast castform = (Form_pg_cast) GETSTRUCT(casttup);
+
+		result = castform->castfunc;
+		ReleaseSysCache(casttup);
+	}
+
+	if (!OidIsValid(result))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("error looking up cast from %d to %d", source, target)));
+
+	return result;
+}
+
+/*
+ * returns true if v1 and v2 reference the same object
+ */
+static bool
+var_equal(Var *v1, Var *v2)
+{
+	return v1->varno == v2->varno && v1->varattno == v2->varattno && v1->vartype == v2->vartype;
+}
+
+static bool
+is_const_null(Expr *expr)
+{
+	return IsA(expr, Const) &&castNode(Const, expr)->constisnull;
+}
+
+static bool
+is_simple_expr_walker(Node *node, void *context)
+{
+	if (node == NULL)
+		return false;
+
+/*
+ * since expression_tree_walker does early exit on true
+ * logic is reverted and return value of true means expression
+ * is not simple, this is reverted in parent
```

Comment:
Ok, formatting checks for the `tsl` _have_ to be broken

Reply:
thats what my pgindent came up with

Reply:
-_- we can't switch to clang-format soon enough